### PR TITLE
[incubator-kie-issues-2363] Migrate EscalationEventTest methods from v7 legacy runtime to code generation

### PR DESCRIPTION
**Pull request by SUNILathulya**
closes https://github.com/apache/incubator-kie-issues/issues/2363

Migrated escalation event handling tests which are being compiled using 
legacy v7 to code generation approach.

The tests can be identified by referring to EscalationEventTest.java:
https://github.com/apache/incubator-kie-kogito-runtimes/blob/main/jbpm/jbpm-tests/src/test/java/org/jbpm/bpmn2/EscalationEventTest.java

## Tests Migrated

**testEscalationBoundaryEventInterrupting()**
- Migrates escalation boundary event on a subprocess
- Uses `EscalationBoundaryEventInterruptingProcess` and `EscalationBoundaryEventInterruptingModel`
- Boundary event is interrupting (`cancelActivity=true`) — cancels the subprocess
- Explicitly completes the `MyTask` work item via `ProcessTestHelper.completeWorkItem()`
- BPMN: `BPMN2-EscalationBoundaryEventInterrupting.bpmn2`

**testInterruptingEscalationBoundaryEventOnTask()**
- Migrates interrupting escalation boundary event on a parallel user task flow
- Uses `EscalationBoundaryEventOnTaskInterruptingProcess` and model
- Parallel gateway splits into john's task (throws escalation) and mary's task (interrupted by boundary)
- Completes john's task first (triggers escalation), then mary's task (interrupted path)
- Replaces `SecurityPolicy.of("john", ...)` with user name string directly in `ProcessTestHelper`
- BPMN: `BPMN2-EscalationBoundaryEventOnTaskInterrupting.bpmn2`

## Migration Changes
- Replaced `createKogitoProcessRuntime()` with generated process classes
- Used `ProcessTestHelper` for application, handler and event listener setup
- Replaced `kruntime.startProcess()` with `processDefinition.createInstance(model).start()`
- Replaced `kruntime.getKogitoWorkItemManager().completeWorkItem()` with `ProcessTestHelper.completeWorkItem()`
- Replaced `SecurityPolicy.of()` with user name string directly
- Replaced `assertProcessInstanceCompleted()` and `assertProcessInstanceFinished()` 
  with direct `processInstance.status()` check

## BPMN Files
- `BPMN2-EscalationBoundaryEventInterrupting.bpmn2` (test 1)
- `BPMN2-EscalationBoundaryEventOnTaskInterrupting.bpmn2` (test 2)

### CODE REVIEW

**Review by pefernan — APPROVED** [review]

File: jbpm/jbpm-tests/src/test/java/org/jbpm/bpmn2/EscalationEventTest.java [hunk 1]
```diff
@@ -186,11 +189,16 @@ public void testEscalationBoundaryEvent() {
...
-        kruntime.getKogitoWorkItemManager().registerWorkItemHandler("MyTask", handler);
-        KogitoProcessInstance processInstance = kruntime.startProcess("EscalationBoundaryEventInterrupting");
-        assertProcessInstanceCompleted(processInstance);
+        ProcessTestHelper.registerHandler(app, "MyTask", handler);
+        org.kie.kogito.process.Process<EscalationBoundaryEventInterruptingModel> processDefinition = EscalationBoundaryEventInterruptingProcess.newProcess(app);
+        org.kie.kogito.process.ProcessInstance<EscalationBoundaryEventInterruptingModel> processInstance = processDefinition.createInstance(processDefinition.createModel());
+        processInstance.start();
+        KogitoWorkItem workItem = handler.getWorkItem();
+        assertThat(workItem).isNotNull();
+        ProcessTestHelper.completeWorkItem(processInstance, Collections.emptyMap(), "john");
```
Comment (martinweiler) [inline jbpm/jbpm-tests/src/test/java/org/jbpm/bpmn2/EscalationEventTest.java:200]: Why do we have to manually complete the work item here? If I look at this process, then the expectation should be that the process gets completed through the interrupting escalation event. That's also what the legacy test does.

File: jbpm/jbpm-tests/src/test/java/org/jbpm/bpmn2/EscalationEventTest.java [hunk 2]
```diff
@@ -253,8 +262,11 @@ public void testInterruptingEscalationBoundaryEventOnTask() throws Exception {
             workItem = workItems.get(1);
         }
 
-        kruntime.getKogitoWorkItemManager().completeWorkItem(workItem.getStringId(), Collections.emptyMap(), SecurityPolicy.of("john", Collections.emptyList()));
-        assertProcessInstanceFinished(processInstance, kruntime);
+        // completing john's task throws escalation which interrupts mary's UserTask via boundary event
+        ProcessTestHelper.completeWorkItem(processInstance, Collections.emptyMap(), "john");
+        // mary's task is interrupted by the boundary event, completing remaining work item
```
Comment (martinweiler) [inline jbpm/jbpm-tests/src/test/java/org/jbpm/bpmn2/EscalationEventTest.java:267]: Similar to the above test, I don't think it should be required to manually complete mary's task that has the escalation event.

File: jbpm/jbpm-tests/src/test/java/org/jbpm/bpmn2/EscalationEventTest.java (hunk 1 above)
Comment (SUNILathulya) [reply jbpm/jbpm-tests/src/test/java/org/jbpm/bpmn2/EscalationEventTest.java:200]: @martinweiler  You are absolutely right — per spec and per the v7 behaviour, this completeWorkItem call should not be needed. The BPMN subprocess has a parallel gateway with two paths: one leads directly to an EscalationEndEvent (no wait state), which should immediately trigger the cancelActivity="true" boundary escalation on the subprocess, cancelling MyTask, and routing through the ScriptTask to completion.

However, in v9 the interrupting escalation does not cancel the pending work item. Removing the completeWorkItem call leaves the process in STATE_ACTIVE (1) instead of STATE_COMPLETED (2). This is not a test bug — it is a known v9 engine limitation, documented by the already-disabled adjacent test testEscalationBoundaryEventInterruptsTask which carries the comment: "Escalation does not cancel work items yet."

The extra completeWorkItem is a deliberate workaround to keep this test exercising the escalation-boundary-event routing path while the v9 engine gap exists.

File: jbpm/jbpm-tests/src/test/java/org/jbpm/bpmn2/EscalationEventTest.java (hunk 2 above)
Comment (SUNILathulya) [reply jbpm/jbpm-tests/src/test/java/org/jbpm/bpmn2/EscalationEventTest.java:267]: @martinweiler  Same root cause as the comment above. In v7, completing john's task was sufficient: the escalation end event fired BoundaryEvent_1 (interrupting, cancelActivity="true") on mary's UserTask_1, which cancelled it automatically, allowing the process to finish.

In v9, the interrupting escalation boundary event does not cancel Mary's pending task. Removing the second completeWorkItem call results in expected: 2 but was: 1 — the process stays active. This is the same missing feature that the testEscalationBoundaryEventInterruptsTask test documents. The second completion is a workaround until v9 implements work-item cancellation via interrupting escalation boundary events.

File: jbpm/jbpm-tests/src/test/java/org/jbpm/bpmn2/EscalationEventTest.java (hunk 1 above)
Comment (martinweiler) [reply jbpm/jbpm-tests/src/test/java/org/jbpm/bpmn2/EscalationEventTest.java:200]: Thanks for the detailed follow-up @SUNILathulya !

Discoveries like this are the true win of this test case conversion task! Note that the verifications and asserts in the test cases should be the same before and after the conversion. If for some reason the test then fails, we need to investigate in that direction as it might indicate that the current engine behavior is incorrect. What we should *not* do is to adapt the test case so that it fits the engine's behavior.

This being said, it seems for these test cases there is another open PR that might resolve the issue: https://github.com/apache/incubator-kie-kogito-runtimes/pull/3312

Now that this PR needs to be re-opened against the drools repository, I'd suggest to combine it with the changes from PR3312.
